### PR TITLE
Added automated way to build documentation and deploy it to github pages

### PR DESCRIPTION
**Pull request by fedem-p**
After some bugs, it can finally be triggered only by a push from the main branch.

### CODE REVIEW

**Comment by vigji** [comment]
This is absolutely great @fedem-p!!!

There seems to be a funny duplications of titles in the left, do you know why that happens?

**Comment by fedem-p** [comment]
I reorganized everything and now it should display correctly the different sections +  the API.
 
todo:
- link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
-  add an index on the homepage (?)

**Comment by diegoasua** [comment]
> 
> 
> I reorganized everything and now it should display correctly the different sections + the API.
> 
> todo:
> 
>     * link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
> 
>     * add an index on the homepage (?)

For the first part, does [this](https://github.com/miyakogi/m2r/issues/41) work? For the second part, there is already a searchable index on the left side when checking the compiled web, and as adding the index to homepage seems to be messy it is probably in your interest to just forget about that.

**Comment by fedem-p** [comment]
> > I reorganized everything and now it should display correctly the different sections + the API.
> > todo:
> > ```
> > * link the readme instead of copying ( I tried a bit but it seems harder than I though..    using rst to link it seems to not work with jupyter books, and I couldn't find any solution using only markdown)
> > 
> > * add an index on the homepage (?)
> > ```
> 
> For the first part, does [this](https://github.com/miyakogi/m2r/issues/41) work? For the second part, there is already a searchable index on the left side when checking the compiled web, and as adding the index to homepage seems to be messy it is probably in your interest to just forget about that. I like the new style of the docs, it feels very clean

File: .github/workflows/doc_deploy.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    # Install dependencies
+    - name: Set up Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: "3.8"
+
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install -r requirements.txt
```
Comment (diegoasua) [inline .github/workflows/doc_deploy.yml:25]: Is there a need to install all [requirements](https://github.com/portugueslab/sashimi/blob/master/requirements.txt) from the main program, to build the docs? Is just toml enough maybe? Or not even, in which case `Install dependencies` job can be skipped altogether.

File: docs/intro.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
+# Sashimi
+
+<a href="url"><img 
+src="https://github.com/portugueslab/sashimi/blob/master/sashimi/icons/main_icon.png"
```
Comment (diegoasua) [inline docs/intro.md:4]: [This image](https://portugueslab.github.io/sashimi/intro.html) does not seem to load correctly. Try without the hyperlink (`<a></a>`) tag:

```html
<img src=https://github.com/portugueslab/sashimi/blob/master/sashimi/icons/main_icon.png align="left" height="190" width="270">
```

File: docs/intro.md [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+
+It is a good practice to create an environment for every project. The provided `environment.yml` sets up almost all required dependencies (see below).
+
+    conda env create -f {path to environment.yml}
+
+You can activate the environment by running:
+
+    conda activate sashimi
+    
+After this you moght have to install two extra dependencies for controlling a Cobolt laser:
```
Comment (diegoasua) [inline docs/intro.md:42]: Technically this should read for controlling a laser through serial communication, such as Cobolt lasers.

**Review by diegoasua — APPROVED** [review]
Awesome work on the sphinx automation in `docs/_templates`. Overall I don't have anything to add and the very few comments on this are just picky things on my side. After they have been addressed feel free to merge to main branch. Good work! 😃

File: docs/index.md [hunk 4]
```diff
@@ -0,0 +1,4 @@
+# Welcome to the Sashimi documentation!
+
+maybe add index here?
```
Comment (vilim) [inline docs/index.md:3]: It should be an internal comment for us, not user-visible

**Review by vilim — APPROVED** [review]
Only one minor comment, otherwise good to merge!

File: .github/workflows/doc_deploy.yml (hunk 1 above)
Comment (fedem-p) [reply .github/workflows/doc_deploy.yml]: You were right! Thanks!

**Comment by fedem-p** [comment]
I've fixed the problem with the image by changing the URL, now with this, it works both on GitHub and the documentation.
I've also restyled the structure a tiny bit and added an index at the home page (that can be removed easily if you don't like it)

Regarding linking the README file directly, I couldn't find anything that worked.  
The problem seems to be the fact that I cannot use the mainstream way to do it with an .rst because it doesn't compile properly with Jupiter books.  There is some workaround to do that just with .md files, but I still need to find one that works.  
In the end, I just copied pasted the content and adjusted the structure to have different sections for installation and configuration.

**Comment by diegoasua** [comment]
And.... Done!

**Pull request merged**